File: lims-web-app/src/services/labTypeService.js
import axios from "axios";
import { Urls } from "./apiURLS";

export const getLabTypes = async (startsWith = '', pageNumber, pageSize, createdBy) => {
  try {
    const response = await axios.post(
      `${Urls.MASTERS}lab-type/get-all?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=name&startsWith=${startsWith}`, {}, {
      headers: {
        createdBy
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error fetching lab types:', error);
    return error;
  }
};

export const addLabType = async (labType, createdBy) => {
  try {
    const response = await axios.post(`${Urls.MASTERS}lab-type/save`, labType, {
      headers: {
        'createdBy': createdBy
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error adding lab type:', error);
    return error;
  }
};

export const updateLabType = async (id, labType, userId) => {
  try {
    const response = await axios.put(`${Urls.MASTERS}lab-type/update/${id}`, labType, {
      headers: {
        'userId': userId
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error updating lab type:', error);
    return error;
  }
};

export const deleteLabType = async (id) => {
  try {
    const response = await axios.delete(`${Urls.MASTERS}lab-type/${id}`);
    return response.data;
  } catch (error) {
    console.error('Error deleting lab type:', error);
    return error;
  }
};

export const getLabTypeById = async (id) => {
  try {
    const response = await axios.get(`${Urls.MASTERS}lab-type/${id}`);
    return response.data;
  } catch (error) {
    console.error('Error fetching lab type by id:', error);
    return error;
  }
};

File: back-end-lims/master-data-service/java/com/digiworldexpo/lims/master/service/LabTypeService.java
package com.digiworldexpo.lims.master.service;

import java.util.List;

import java.util.UUID;

import com.digiworldexpo.lims.entities.master.LabType;
import com.digiworldexpo.lims.master.model.response.ResponseModel;

public interface LabTypeService {

	ResponseModel<LabType> saveLabType(LabType labType, UUID createdBy);

	ResponseModel<List<LabType>> getLabTypes(String startsWith, int pageNumber, int pageSize, String sortBy,
			UUID createdBy);

	ResponseModel<LabType> getLabTypeById(UUID id);

	ResponseModel<LabType> updateLabType(UUID id, LabType updatedLabType, UUID userId);

	ResponseModel<LabType> deleteLabType(UUID id);
}
